lib/awsv4/header.js

```diff
@@ -19,33 +19,42 @@ function fillOptions(options) {
 function signWithHeaders(method, service, region = "", target = "", path = "", body = null, query = "", headers = {}) {
   var options = { headers: headers }
   options = fillOptions(options);
+  options.headers["X-Amz-Date"] = toTime(options.timestamp)
+  var host = service.concat(".", options.region, ".amazonaws.com")
+  options.headers["Host"] = host
+
+  if (options.sessionToken) {
+    options.headers["X-Amz-Security-Token"] = options.sessionToken
+  }
 
   if (target) {
     options.headers["X-Amz-Target"] = target;
   }
 
+  var svc = service.split(".").pop()
   var credential = options.key +
     "/" +
-    createCredentialScope(options.timestamp, region, service);
+    createCredentialScope(options.timestamp, options.region, svc);
```

Comment:
This is an example of what I mean when I say the API still needs to be ironed out. Sagemaker URLs use `runtime.sagemaker.${REGION}.amazonaws.com`, but the service name for the purposes of request signing is simply `sagemaker`.

Reply:
I haven't looked at every AWS service and how it handles it, but IMO this should just be one more parameter:
1. we have the `service` which is what `svc` is here 
2. `serviceSubdomain` which if not set is just equal to `service`

And then for the services that have this be different the user will need to provide both or do a wrapper, that can then be added to the jslib ;)